### PR TITLE
HTTP SSL support

### PR DESCRIPTION
**Pull request by istiak101**
Added Gin SSL support

### CODE REVIEW

File: pkg/command/options.go [hunk 1]
```diff
@@ -41,6 +41,9 @@ type Options struct {
 	RetryCount                   uint   `long:"open-retry" description:"Number of times to retry establishing connection" default:"0"`
 	HTTPHost                     string `long:"bind" description:"HTTP server host" default:"localhost"`
 	HTTPPort                     uint   `long:"listen" description:"HTTP server listen port" default:"8081"`
+	HTTPSSLMode                  bool   `long:"httpssl" description:"HTTP server SSL Mode"`
```
Comment (sosedoff) [inline pkg/command/options.go:44]: I think its best if we specify the mode via `TLS` (--tls) attribute instead.  You'll end up with:

```
--tls (enables the TLS)
--tls-cert path
--tls-key path
```

**Comment by sosedoff** [comment]
A few questions:

- Will this work with private key with password?
- Can i use self-signed certs?

**Comment by istiak101** [comment]
> A few questions:
> 
> * Will this work with private key with password?
> * Can i use self-signed certs?

1. Private key with password : No.
2. Self signed certificate: Yes

**Comment by sosedoff** [comment]
Can you provide instructions on how to run pgweb in TLS mode with self signed cert? I can put that information into WIKI

**Comment by istiak101** [comment]
> Can you provide instructions on how to run pgweb in TLS mode with self signed cert? I can put that information into WIKI

No extra steps needed. Only file path given with flags is enough for both self-signed and trusted certificate.

**Comment by sosedoff** [comment]
@istiak101 i meant more like do you have commands to generate the self-signed cert that i can put into wiki so folks dont have to search troves of online tutorials

**Comment by istiak101** [comment]
Ah I misunderstood.

`openssl req -x509 -newkey ec -pkeyopt ec_paramgen_curve:secp384r1 -days 365
  -nodes -keyout example.com.key -out example.com.crt -subj "/CN=localhost" -addext "subjectAltName=DNS:localhost,DNS:*.localhost"`

**Comment by sosedoff** [comment]
I tried running these changes locally. Before we merge the PR we need to ensure we're validating the flags, and checking if cert/key files exist. Something along these lines:

```diff
diff --git a/pkg/command/options.go b/pkg/command/options.go
index b1e34b8..6870a01 100644
--- a/pkg/command/options.go
+++ b/pkg/command/options.go
@@ -198,6 +198,17 @@ func ParseOptions(args []string) (Options, error) {
 		}
 	}
 
+	if opts.TLSMode {
+		if opts.TLSCert == "" {
+			return opts, errors.New("tls cert is not provided")
+		}
+		if opts.TLSKey == "" {
+			return opts, errors.New("tls key is not provided")
+		}
+
+		// also need to validate if these files exists
+	}
+
 	return opts, nil
 }
 
```

There's `options_test.go` file that can help with that.

**Comment by sosedoff** [comment]
@istiak101 why close this PR?

**Pull request closed without merging**